cirq-core/cirq/ops/fsim_gate.py

```diff
@@ -90,8 +90,32 @@ def __init__(self, theta: float, phi: float) -> None:
                 ``|11⟩`` state is phased. Note: uses opposite sign convention to
                 the CZPowGate. Maximum strength (full cz) is at pi.
         """
-        self.theta = _canonicalize(theta)
-        self.phi = _canonicalize(phi)
+        self._theta = _canonicalize(theta)
+        self._phi = _canonicalize(phi)
+
+    @property
+    def theta(self) -> Union[float, sympy.Basic]:
+        return self._theta
```

Comment:
Return type hints here don't match the ones from `__init__` should we fix that ?

Reply:
It matches the return type from `_canonicalize`, which is what the member is actually set to in the constructor.

Reply:
Also I don't have merge access.

Reply:
So looking at:

```python3
def _canonicalize(value: Union[float, sympy.Basic]) -> Union[float, sympy.Basic]:
```
And 
```python3
def __init__(self, theta: float, phi: float) -> None:
```
Why shouldn't we update it to this:
```python3
def __init__(self, theta: Union[float, sympy.Basic], phi: Union[float, sympy.Basic]) -> None:
```
because just as a float the type hints are narrower than they need to be no ?

Reply:
Ah, I thought you meant to go the other way (which would be perfectly viable too since the implementation of `_canonicalize` would always return a `float` in this case).

I didn't want to expand the input types because this class may not actually support parameters. But looking at the rest of the class, it looks like parameters are supported. So I'll go ahead and update the `__init__` sig.

FWIW `TParamVal` is an alias for that union, so I'll use that everywhere instead.